lib/cctp/mint.go

```diff
@@ -260,11 +262,11 @@ func didReceive(ctx context.Context, msgTransmitter *MessageTransmitter, msg typ
 	//  recipient             32         bytes32    52
 	//  messageBody           dynamic    bytes      84
 	//
-	// Nonce key is keccak256(abi.encodePacked(nonce, sourceDomain))
-	nonceKey := crypto.Keccak256Hash(append(
-		msg.MessageBytes[12:20],  // nonce
-		msg.MessageBytes[4:8]..., // source domain
-	))
+	// Nonce key is keccak256(abi.encodePacked(sourceDomain, nonce))
+	var nonceBz []byte
+	nonceBz = append(nonceBz, msg.MessageBytes[4:8]...)   // source domain
+	nonceBz = append(nonceBz, msg.MessageBytes[12:20]...) // nonce
+	nonceKey := crypto.Keccak256Hash(nonceBz)
```

Comment:
this was a nasty one. was mutating message bytes 👎  i blame ai

Reply:
yeah, only ever do `x := append(x, <stuff here>)`, never do `y := append(x, <stuff here>`